### DISCUSSION TITLE
Clarify the role of `interpolation_steps`?

### DISCUSSION THREAD

**Discussion by peter-mitrano-bg** [NVlabs/curobo#258] [Research Q&A] [answered]
From the docstring, it would seem like this mean that the shape of the interpolated plan is always exactly `interpolation_steps` which defaults to `5000`. However, that wouldn't make sense, since fine-tuning can change the dt and there also seems to be a fixed interpolation dt that takes priority. I'm seeing that my interpolated trajectories are not in fact 500 long, but vary in length depending on the query.

Could we clarify how this parameter works? When is it respected or ignored? From reading the code it really only seems to effect the graph planner?

**Comment by balakumar-s** [comment, marked as answer]
`interpolation_steps` is a max buffer to store interpolated trajectories in. We use a CUDA kernel to interpolate batch of trajectories given solutions from the solver. Hence we need to preallocate tensors for the interpolated trajectories. We additionally store the last index for each interpolated trajectory in the batch and use this index to return variable length interpolated trajectories through `MotionGenResult.get_interpolated_plan()`.

**Reply by balakumar-s** [reply]
A better name would be `max_interpolation_cache`.
